cmd/sos_show.go

```diff
@@ -0,0 +1,187 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/exoscale/cli/table"
+	"github.com/spf13/cobra"
+)
+
+type sosACLShowOutput struct {
+	Access string `json:"access"`
+	Value  string `json:"value"`
+}
+
+type sosMetadataShowOutput struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type sosHeadersShowOutput struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type sosShowOutput struct {
+	Object   string                  `json:"object"`
+	URL      string                  `json:"url"`
+	ACL      []sosACLShowOutput      `json:"acl"`
+	Metadata []sosMetadataShowOutput `json:"metadata"`
+	Headers  []sosHeadersShowOutput  `json:"headers"`
+}
+
+func (o *sosShowOutput) toJSON() { outputJSON(o) }
+
+func (o *sosShowOutput) toText() { outputText(o) }
+
+func (o *sosShowOutput) toTable() {
+	t := table.NewTable(os.Stdout)
+	t.SetHeader([]string{o.Object})
```

Comment:
I'm not sure we should add a header with the object name:

* it can result in a veeeeeeeeeeeeeeeeeeeeeery wide column if the object name is veeeeeeeeeeeeeeeeeeeeeery long
* according to your example, the displayed object is not exactly the actual object name (underscores are replaced with whitespaces, and so will be other characters I guess...)

Thus, I suggest not putting a table header and instead add an `Object | <the object name>` row in the table.

Reply:
I think we don't even need to display it

Reply:
Yes, that was my point 😉